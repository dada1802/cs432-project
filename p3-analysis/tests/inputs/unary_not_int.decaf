def int main() {
    int a;
    bool b;
    a = !4;
    return 0;
}
