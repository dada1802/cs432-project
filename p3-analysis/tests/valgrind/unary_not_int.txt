==136787== Memcheck, a memory error detector
==136787== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==136787== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==136787== Command: ../decaf inputs/unary_not_int.decaf
==136787== 
Type mismatch: bool expected but int found on line 4
Type mismatch: int is incompatible with bool on line 4
==136787== 
==136787== HEAP SUMMARY:
==136787==     in use at exit: 0 bytes in 0 blocks
==136787==   total heap usage: 1,867 allocs, 1,867 frees, 226,376 bytes allocated
==136787== 
==136787== All heap blocks were freed -- no leaks are possible
==136787== 
==136787== For lists of detected and suppressed errors, rerun with: -s
==136787== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
