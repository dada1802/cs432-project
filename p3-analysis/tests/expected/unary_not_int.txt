Type mismatch: bool expected but int found on line 4
Type mismatch: int is incompatible with bool on line 4
